### ISSUE TITLE
GNU/Linux memory dump: A translation table/symbol table requirement was not fulfilled.

### ISSUE THREAD

**Issue by Necklaces** [volatilityfoundation/volatility3#740]
Sorry for ignoring most of the bug reporting template, I know there are a couple of similar issues like this, but stick with me here will ya.

I've been struggling with another dump for a while and decided: "let's do it in reverse", so here's what I did:

- Volatility distributes sample GNU/Linux symbol tables (https://downloads.volatilityfoundation.org/volatility3/symbols/linux.zip). In that zip there's a CentOS sample: centos-2.6.18-8.1.15.el5.json.xz. I found out that this specific CentOS kernel is only used in CentOS 5.0. If we have a closer look in the sample, that specific sample seems to be from an i386 system and not an x86_64 system.

So, theoretically, if I set up a CentOS 5.0 i386 VM and get a memory dump from that, volatility3 should use this symbol table and I should get to run linux plugins like pslist, correct?

Well, that's what I did, but unfortunately volatility3 still doesn't work!

The symbol table, aka the isf file, has this banner:
```
$ vol isfinfo --isf centos-2.6.18-8.1.15.el5.json.xz
[...]
Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007
```

My images (.raw, .lime, .vmem) all have the same banner!

```
$ vol -f memory_dump_using_lime_1.1-r17.lime banners.Banner
[...]
0x600000	Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007
[...]
```

That is the *exact* same banner. I ran a diff twice just to be sure.

Yet, when I try to run a simple pslist, this is what I get:

```
$ vol -f memory_dump_using_lime_1.1-r17.lime -s centos-2.6.18-8.1.15.el5.json.xz linux.pslist.PsList
Volatility 3 Framework 2.2.0
Progress:  100.00		Stacking attempts finished
Unsatisfied requirement plugins.PsList.kernel.layer_name: 
Unsatisfied requirement plugins.PsList.kernel.symbol_table_name: 

A translation layer requirement was not fulfilled.  Please verify that:
	A file was provided to create this layer (by -f, --single-location or by config)
	The file exists and is readable
	The file is a valid memory image and was acquired cleanly

A symbol table requirement was not fulfilled.  Please verify that:
	The associated translation layer requirement was fulfilled
	You have the correct symbol file for the requirement
	The symbol file is under the correct directory or zip file
	The symbol file is named appropriately or contains the correct banner

Unable to validate the plugin requirements: ['plugins.PsList.kernel.layer_name', 'plugins.PsList.kernel.symbol_table_name']
```

<details>
  <summary>EXPANDABLE SECTION: Full -vvvvv verbose output </summary>

```
$ vol -vvvvv -s centos-2.6.18-8.1.15.el5.json.xz -f memory_dump_using_lime_1.1-r17.lime linux.pslist.PsList
Volatility 3 Framework 2.2.0
INFO     volatility3.cli: Volatility plugins path: ['/usr/lib/python3.10/site-packages/volatility3/plugins', '/usr/lib/python3.10/site-packages/volatility3/framework/plugins']
INFO     volatility3.cli: Volatility symbols path: ['~/centos/centos-2.6.18-8.1.15.el5.json.xz', '/usr/lib/python3.10/site-packages/volatility3/symbols', '/usr/lib/python3.10/site-packages/volatility3/framework/symbols']
Level 7  volatility3.cli: Cache directory used: ~/.cache/volatility3
INFO     volatility3.framework.automagic: Detected a linux category plugin
INFO     volatility3.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
INFO     volatility3.framework.automagic: Running automagic: SymbolBannerCache
INFO     volatility3.framework.automagic: Running automagic: LinuxBannerCache
INFO     volatility3.framework.automagic.symbol_cache: Building linux caches...
Level 7  volatility3.framework.layers.resources: Available URL handlers: HTTPErrorProcessor, HTTPDefaultErrorHandler, HTTPRedirectHandler, ProxyHandler, HTTPBasicAuthHandler, ProxyBasicAuthHandler, HTTPDigestAuthHandler, ProxyDigestAuthHandler, AbstractHTTPHandler, HTTPHandler, HTTPSHandler, HTTPCookieProcessor, UnknownHandler, FileHandler, FTPHandler, CacheFTPHandler, DataHandler, VolatilityHandler, JarHandler, OfflineHandler
INFO     volatility3.framework.automagic: Running automagic: LayerStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Stacked LimeLayer using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LinuxIntelStacker
DEBUG    volatility3.framework.automagic.linux: Identified banner: b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00'
Level 7  volatility3.framework.automagic.stacker: Exception during stacking: Symbol type not in LintelStacker1 SymbolTable: mnt_namespace
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: LimeLayer
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: FileLayer
DEBUG    volatility3.framework.automagic.stacker: Stacked layers: ['LimeLayer', 'FileLayer']
INFO     volatility3.framework.automagic: Running automagic: SymbolFinder  
INFO     volatility3.framework.automagic: Running automagic: LinuxSymbolFinder
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
INFO     volatility3.framework.automagic: Running automagic: KernelModule
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name

Unsatisfied requirement plugins.PsList.kernel.layer_name: 
Unsatisfied requirement plugins.PsList.kernel.symbol_table_name: 

A translation layer requirement was not fulfilled.  Please verify that:
	A file was provided to create this layer (by -f, --single-location or by config)
	The file exists and is readable
	The file is a valid memory image and was acquired cleanly

A symbol table requirement was not fulfilled.  Please verify that:
	The associated translation layer requirement was fulfilled
	You have the correct symbol file for the requirement
	The symbol file is under the correct directory or zip file
	The symbol file is named appropriately or contains the correct banner

Unable to validate the plugin requirements: ['plugins.PsList.kernel.layer_name', 'plugins.PsList.kernel.symbol_table_name']
```  

</details>

Now I know what you''re gonna say "maybe you did something wrong", and I admit I might've - I had to find a pretty old version of LiME because GCC 4.1.1 wouldn't build any of the releases on GitHub so eventually I found 1.1-r17 and maybe there is a problem with that. I didn't test AVML but I doubt that would work on such an old kernel anyway.

Which is why I also took the .vmem file! And that doesn't work either. Same problem. So it's not the old lime version. 

Anyway, I uploaded my freshly baked memory samples to archive.org so you can check them for yourselves: 

- [CentOS 5.0 i386 2.6.18-8.1.15.el5 memory samples](https://archive.org/details/20220520_20220520_1840)

I hope this helps in solving this issue - unless I screwed something up and it actually does work. I tried making a config, but failed hard, and I don't know volshell so I'm depending on automagic..

Either way, thank you for all your hard work. Volatility is a great tool to have!

**Context**
Volatility Version:  v2.0.1.3291.9bfa7187
Operating System:  Arch GNU/Linux
Python Version:  3.10.4
Suspected Operating System:  CentOS 5.0 i386
Command:  vol -f memory_dump_using_lime_1.1-r17.lime -s centos-2.6.18-8.1.15.el5.json.xz linux.pslist.PsList

**To Reproduce**

1. Download one of my memory samples
2. Run any command using provided isf / symbol table
3. See error

**Expected behavior**
It shouldn't complain about translation table or symbol table when I have them.

**Comment by Necklaces** [comment]
Okay so I got it working, but I'm a little unsure if there's been a regression in one of the latest commits after release 2.0.1 or not.

- 1.0. Using latest build -  v2.0.1.3291.9bfa7187 (prints 2.2.0 in output) it never works:

- 1.1. Trying to give the symbols, or the folder with the symbols, directly - does not work.

<details>
  <summary>Symbols with -s</summary>

```
$ vol --version
Volatility 3 Framework 2.2.0
$ sudo cp linux.zip /usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip
$ vol -vvvvv -s ~/centos/ -f memory_dump_using_lime_1.1-r17.lime linux.pslist.PsList
Volatility 3 Framework 2.2.0
INFO     volatility3.cli: Volatility plugins path: ['/usr/lib/python3.10/site-packages/volatility3/plugins', '/usr/lib/python3.10/site-packages/volatility3/framework/plugins']
INFO     volatility3.cli: Volatility symbols path: ['~/centos', '/usr/lib/python3.10/site-packages/volatility3/symbols', '/usr/lib/python3.10/site-packages/volatility3/framework/symbols']
Level 7  volatility3.cli: Cache directory used: ~/.cache/volatility3
INFO     volatility3.framework.automagic: Detected a linux category plugin
INFO     volatility3.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
INFO     volatility3.framework.automagic: Running automagic: SymbolBannerCache
INFO     volatility3.framework.automagic: Running automagic: LinuxBannerCache
INFO     volatility3.framework.automagic.symbol_cache: Building linux caches...
Level 7  volatility3.framework.layers.resources: Available URL handlers: HTTPErrorProcessor, HTTPDefaultErrorHandler, HTTPRedirectHandler, ProxyHandler, HTTPBasicAuthHandler, ProxyBasicAuthHandler, HTTPDigestAuthHandler, ProxyDigestAuthHandler, AbstractHTTPHandler, HTTPHandler, HTTPSHandler, HTTPCookieProcessor, UnknownHandler, FileHandler, FTPHandler, CacheFTPHandler, DataHandler, VolatilityHandler, JarHandler, OfflineHandler
INFO     volatility3.framework.automagic: Running automagic: LayerStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Stacked LimeLayer using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LinuxIntelStacker
INFO     volatility3.framework.automagic.linux: No Linux banners found - if this is a linux plugin, please check your symbol files location
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: LimeLayer
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: FileLayer
DEBUG    volatility3.framework.automagic.stacker: Stacked layers: ['LimeLayer', 'FileLayer']
INFO     volatility3.framework.automagic: Running automagic: SymbolFinder
INFO     volatility3.framework.automagic: Running automagic: LinuxSymbolFinder
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
INFO     volatility3.framework.automagic: Running automagic: KernelModule
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name

Unsatisfied requirement plugins.PsList.kernel.layer_name: 
Unsatisfied requirement plugins.PsList.kernel.symbol_table_name: 

A translation layer requirement was not fulfilled.  Please verify that:
	A file was provided to create this layer (by -f, --single-location or by config)
	The file exists and is readable
	The file is a valid memory image and was acquired cleanly

A symbol table requirement was not fulfilled.  Please verify that:
	The associated translation layer requirement was fulfilled
	You have the correct symbol file for the requirement
	The symbol file is under the correct directory or zip file
	The symbol file is named appropriately or contains the correct banner

Unable to validate the plugin requirements: ['plugins.PsList.kernel.layer_name', 'plugins.PsList.kernel.symbol_table_name']
```
</details>

- 1.2. Putting symbols in /usr/lib/python3.10/site-packages/volatility3/symbols/ - also does not work.
    
<details>
  <summary>Linux.zip in /symbols</summary>

```
$ vol --version
Volatility 3 Framework 2.2.0

vol -vvvvv -f memory_dump_using_lime_1.1-r17.lime linux.pslist.PsList
Volatility 3 Framework 2.2.0
INFO     volatility3.cli: Volatility plugins path: ['/usr/lib/python3.10/site-packages/volatility3/plugins', '/usr/lib/python3.10/site-packages/volatility3/framework/plugins']
INFO     volatility3.cli: Volatility symbols path: ['/usr/lib/python3.10/site-packages/volatility3/symbols', '/usr/lib/python3.10/site-packages/volatility3/framework/symbols']
Level 7  volatility3.cli: Cache directory used: ~/.cache/volatility3
INFO     volatility3.framework.automagic: Detected a linux category plugin
INFO     volatility3.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
INFO     volatility3.framework.automagic: Running automagic: SymbolBannerCache
INFO     volatility3.framework.automagic: Running automagic: LinuxBannerCache
INFO     volatility3.framework.automagic.symbol_cache: Building linux caches...
Level 7  volatility3.framework.layers.resources: Available URL handlers: HTTPErrorProcessor, HTTPDefaultErrorHandler, HTTPRedirectHandler, ProxyHandler, HTTPBasicAuthHandler, ProxyBasicAuthHandler, HTTPDigestAuthHandler, ProxyDigestAuthHandler, AbstractHTTPHandler, HTTPHandler, HTTPSHandler, HTTPCookieProcessor, UnknownHandler, FileHandler, FTPHandler, CacheFTPHandler, DataHandler, VolatilityHandler, JarHandler, OfflineHandler
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 4.9.0-3-amd64 (debian-kernel@lists.debian.org) (gcc version 6.3.0 20170516 (Debian 6.3.0-18) ) #1 SMP Debian 4.9.30-2+deb9u2 (2017-06-26)\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-4.9.0-3-amd64-dbg_4.9.30-2+deb9u2_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 2.6.32-5-amd64 (Debian 2.6.32-48squeeze6) (jmm@debian.org) (gcc version 4.3.5 (Debian 4.3.5-4) ) #1 SMP Tue May 13 16:34:35 UTC 2014\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-2.6.32-5-amd64-dbg_2.6.32-48squeeze6_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 3.2.0-4-amd64 (debian-kernel@lists.debian.org) (gcc version 4.6.3 (Debian 4.6.3-14) ) #1 SMP Debian 3.2.57-3+deb7u2\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-3.2.0-4-amd64-dbg_3.2.57-3+deb7u2_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/centos-2.6.18-8.1.15.el5.json.xz
INFO     volatility3.framework.automagic: Running automagic: LayerStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Stacked LimeLayer using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LinuxIntelStacker
DEBUG    volatility3.framework.automagic.linux: Identified banner: b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00'
WARNING  volatility3.framework.automagic.linux: Multiple symbol files identified (using *):
WARNING  volatility3.framework.automagic.linux:  * jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/centos-2.6.18-8.1.15.el5.json.xz
WARNING  volatility3.framework.automagic.linux:    jar:file:/usr/lib/python3.10/site-packages/volatility3/framework/symbols/linux.zip!linux/centos-2.6.18-8.1.15.el5.json.xz
Level 7  volatility3.framework.automagic.stacker: Exception during stacking: Symbol type not in LintelStacker1 SymbolTable: mnt_namespace
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: LimeLayer
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: FileLayer
DEBUG    volatility3.framework.automagic.stacker: Stacked layers: ['LimeLayer', 'FileLayer']
INFO     volatility3.framework.automagic: Running automagic: SymbolFinder  
INFO     volatility3.framework.automagic: Running automagic: LinuxSymbolFinder
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
INFO     volatility3.framework.automagic: Running automagic: KernelModule
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name

Unsatisfied requirement plugins.PsList.kernel.layer_name: 
Unsatisfied requirement plugins.PsList.kernel.symbol_table_name: 

A translation layer requirement was not fulfilled.  Please verify that:
	A file was provided to create this layer (by -f, --single-location or by config)
	The file exists and is readable
	The file is a valid memory image and was acquired cleanly

A symbol table requirement was not fulfilled.  Please verify that:
	The associated translation layer requirement was fulfilled
	You have the correct symbol file for the requirement
	The symbol file is under the correct directory or zip file
	The symbol file is named appropriately or contains the correct banner

Unable to validate the plugin requirements: ['plugins.PsList.kernel.layer_name', 'plugins.PsList.kernel.symbol_table_name']
```
</details>

- 2.0. Using 2.0.1 release it works with symbols in /symbols !

- 2.1. Trying to give the symbols, or the folder with the symbols, directly - does not work.

<details>
  <summary>Symbols with -s</summary>

```
$ vol --version
Volatility 3 Framework 2.0.1
$ vol -vvvvv -s centos-2.6.18-8.1.15.el5.json.xz -f memory_dump_using_lime_1.1-r17.lime linux.pslist.PsList
Volatility 3 Framework 2.0.1
INFO     volatility3.cli: Volatility plugins path: ['/usr/lib/python3.10/site-packages/volatility3/plugins', '/usr/lib/python3.10/site-packages/volatility3/framework/plugins']
INFO     volatility3.cli: Volatility symbols path: ['~/centos/centos-2.6.18-8.1.15.el5.json.xz', '/usr/lib/python3.10/site-packages/volatility3/symbols', '/usr/lib/python3.10/site-packages/volatility3/framework/symbols']
Level 7  volatility3.cli: Cache directory used: ~/.cache/volatility3
INFO     volatility3.framework.automagic: Detected a linux category plugin
INFO     volatility3.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
INFO     volatility3.framework.automagic: Running automagic: LinuxBannerCache
INFO     volatility3.framework.automagic.symbol_cache: Building linux caches...
Level 7  volatility3.framework.layers.resources: Available URL handlers: HTTPErrorProcessor, HTTPDefaultErrorHandler, HTTPRedirectHandler, ProxyHandler, HTTPBasicAuthHandler, ProxyBasicAuthHandler, HTTPDigestAuthHandler, ProxyDigestAuthHandler, AbstractHTTPHandler, HTTPHandler, HTTPSHandler, HTTPCookieProcessor, UnknownHandler, FileHandler, FTPHandler, CacheFTPHandler, DataHandler, VolatilityHandler, JarHandler, OfflineHandler
INFO     volatility3.framework.automagic: Running automagic: LayerStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Stacked LimeLayer using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LinuxIntelStacker
INFO     volatility3.framework.automagic.linux: No Linux banners found - if this is a linux plugin, please check your symbol files location
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: LimeLayer
Level 9  volatility3.framework.configuration.requirements: TypeError - Layer is not the required Architecture: FileLayer
DEBUG    volatility3.framework.automagic.stacker: Stacked layers: ['LimeLayer', 'FileLayer']
INFO     volatility3.framework.automagic: Running automagic: LinuxSymbolFinder
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
INFO     volatility3.framework.automagic: Running automagic: KernelModule
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel

Unsatisfied requirement plugins.PsList.kernel: Linux kernel
Unable to validate the plugin requirements: ['plugins.PsList.kernel']
```

</details>

- 2.2. Putting symbols in /usr/lib/python3.10/site-packages/volatility3/symbols/ - works!
    
<details>
  <summary>Linux.zip in /symbols</summary>

```
$ vol --version
Volatility 3 Framework 2.0.1
$ sudo cp symbols.zip /usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip
vol -vvvvv -f memory_dump_using_lime_1.1-r17.lime linux.pslist.PsList
Volatility 3 Framework 2.0.1
INFO     volatility3.cli: Volatility plugins path: ['/usr/lib/python3.10/site-packages/volatility3/plugins', '/usr/lib/python3.10/site-packages/volatility3/framework/plugins']
INFO     volatility3.cli: Volatility symbols path: ['~/centos', '/usr/lib/python3.10/site-packages/volatility3/symbols', '/usr/lib/python3.10/site-packages/volatility3/framework/symbols']
Level 7  volatility3.cli: Cache directory used: ~/.cache/volatility3
INFO     volatility3.framework.automagic: Detected a linux category plugin
INFO     volatility3.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
INFO     volatility3.framework.automagic: Running automagic: LinuxBannerCache
INFO     volatility3.framework.automagic.symbol_cache: Building linux caches...
Level 7  volatility3.framework.layers.resources: Available URL handlers: HTTPErrorProcessor, HTTPDefaultErrorHandler, HTTPRedirectHandler, ProxyHandler, HTTPBasicAuthHandler, ProxyBasicAuthHandler, HTTPDigestAuthHandler, ProxyDigestAuthHandler, AbstractHTTPHandler, HTTPHandler, HTTPSHandler, HTTPCookieProcessor, UnknownHandler, FileHandler, FTPHandler, CacheFTPHandler, DataHandler, VolatilityHandler, JarHandler, OfflineHandler
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 4.9.0-3-amd64 (debian-kernel@lists.debian.org) (gcc version 6.3.0 20170516 (Debian 6.3.0-18) ) #1 SMP Debian 4.9.30-2+deb9u2 (2017-06-26)\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-4.9.0-3-amd64-dbg_4.9.30-2+deb9u2_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 2.6.32-5-amd64 (Debian 2.6.32-48squeeze6) (jmm@debian.org) (gcc version 4.3.5 (Debian 4.3.5-4) ) #1 SMP Tue May 13 16:34:35 UTC 2014\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-2.6.32-5-amd64-dbg_2.6.32-48squeeze6_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 3.2.0-4-amd64 (debian-kernel@lists.debian.org) (gcc version 4.6.3 (Debian 4.6.3-14) ) #1 SMP Debian 3.2.57-3+deb7u2\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/linux-image-3.2.0-4-amd64-dbg_3.2.57-3+deb7u2_amd64.json.xz
Level 8  volatility3.framework.automagic.symbol_cache: Caching banner b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00' for file jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/centos-2.6.18-8.1.15.el5.json.xz
INFO     volatility3.framework.automagic: Running automagic: LayerStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Stacked LimeLayer using LimeStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using LinuxIntelStacker
DEBUG    volatility3.framework.automagic.linux: Identified banner: b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00'
DEBUG    volatility3.schemas: Validating JSON against schema...
DEBUG    volatility3.schemas: JSON validated against schema (result cached)
DEBUG    volatility3.framework.symbols: Unresolved reference: LintelStacker1!map_segment
DEBUG    volatility3.framework.symbols: Unresolved reference: LintelStacker1!nfs4_lock_state
DEBUG    volatility3.framework.symbols: Unresolved reference: LintelStacker1!nlm_lockowner
DEBUG    volatility3.framework.automagic.linux: Linux ASLR shift values determined: physical 0 virtual 0
DEBUG    volatility3.framework.automagic.linux: DTB was found at: 0x712000
Level 8  volatility3.framework.automagic.stacker: Stacked IntelLayer using LinuxIntelStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using Elf64Stacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using QemuStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using AVMLStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using WindowsCrashDumpStacker
Level 8  volatility3.framework.automagic.stacker: Attempting to stack using VmwareStacker
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name.memory_layer
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel.layer_name.memory_layer.base_layer
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList.kernel
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel
Level 9  volatility3.framework.automagic.construct_layers: Failed on requirement: plugins.PsList
DEBUG    volatility3.framework.automagic.stacker: Stacked layers: ['IntelLayer', 'LimeLayer', 'FileLayer']
INFO     volatility3.framework.automagic: Running automagic: LinuxSymbolFinder
Level 9  volatility3.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.PsList.kernel.symbol_table_name
DEBUG    volatility3.framework.automagic.symbol_finder: Identified banner: b'Linux version 2.6.18-8.1.15.el5 (mockbuild@builder6.centos.org) (gcc version 4.1.1 20070105 (Red Hat 4.1.1-52)) #1 SMP Mon Oct 22 08:32:04 EDT 2007\n\x00'
DEBUG    volatility3.framework.automagic.symbol_finder: Using symbol library: jar:file:/usr/lib/python3.10/site-packages/volatility3/symbols/linux.zip!linux/centos-2.6.18-8.1.15.el5.json.xz
INFO     volatility3.framework.automagic: Running automagic: KernelModule
Level 9  volatility3.framework.configuration.requirements: IndexError - No configuration provided: plugins.PsList.kernel

PID	PPID	COMM
DEBUG    volatility3.framework.symbols: Unresolved reference: symbol_table_name1!map_segment
DEBUG    volatility3.framework.symbols: Unresolved reference: symbol_table_name1!nfs4_lock_state
DEBUG    volatility3.framework.symbols: Unresolved reference: symbol_table_name1!nlm_lockowner

1	0	init
2	1	migration/0
3	1	ksoftirqd/0
4	1	watchdog/0
5	1	migration/1
6	1	ksoftirqd/1
7	1	watchdog/1
8	1	migration/2
9	1	ksoftirqd/2
10	1	watchdog/2
11	1	migration/3
12	1	ksoftirqd/3
13	1	watchdog/3
14	1	events/0
15	1	events/1
16	1	events/2
17	1	events/3
18	1	khelper
19	1	kthread
25	19	kblockd/0
26	19	kblockd/1
27	19	kblockd/2
28	19	kblockd/3
29	19	kacpid
188	19	cqueue/0
189	19	cqueue/1
190	19	cqueue/2
191	19	cqueue/3
194	19	khubd
196	19	kseriod
275	19	pdflush
276	19	pdflush
277	19	kswapd0
278	19	aio/0
279	19	aio/1
280	19	aio/2
281	19	aio/3
496	19	kpsmoused
560	19	scsi_eh_0
576	19	kjournald
609	19	kauditd
643	1	udevd
1323	19	kgameportd
2129	19	kmirrord
2153	19	kjournald
2616	19	vmmemctl
2975	1	vmtoolsd
3021	1	VGAuthService
3477	1	dhclient
3562	1	restorecond
3574	1	auditd
3576	3574	python
3590	1	syslogd
3593	1	klogd
3605	1	irqbalance
3621	1	mcstransd
3634	1	portmap
3655	1	rpc.statd
3691	1	rpc.idmapd
3710	1	dbus-daemon
3767	1	pcscd
3801	1	automount
3820	1	acpid
3831	1	cupsd
3845	1	sshd
3865	1	sendmail
3874	1	sendmail
3886	1	gpm
3897	1	crond
3928	1	xfs
3939	1	anacron
3949	1	atd
3971	1	yum-updatesd
3983	1	avahi-daemon
3984	3983	avahi-daemon
3995	1	hald
3996	3995	hald-runner
4002	3996	hald-addon-acpi
4010	3996	hald-addon-keyb
4022	3996	hald-addon-stor
4024	3996	hald-addon-stor
4097	1	smartd
4103	1	mingetty
4104	1	mingetty
4105	1	mingetty
4106	1	mingetty
4107	1	mingetty
4108	1	mingetty
4109	1	gdm-binary
4170	4109	gdm-binary
4174	1	gdm-binary
4175	4170	Xorg
4223	4170	gnome-session
4274	4223	ssh-agent
4277	1	dbus-launch
4278	1	dbus-daemon
4285	1	gconfd-2
4288	1	gnome-keyring-d
4290	1	gnome-settings-
4304	1	metacity
4310	1	gnome-panel
4312	1	nautilus
4317	1	gnome-vfs-daemo
4318	1	eggcups
4319	1	gnome-volume-ma
4323	1	bonobo-activati
4332	1	bt-applet
4343	1	puplet
4345	1	escd
4351	4345	netstat
4357	1	nm-applet
4364	1	pam-panel-icon
4370	1	gnome-power-man
4379	4364	pam_timestamp_c
4382	1	gam_server
4391	1	wnck-applet
4397	1	trashapplet
4407	1	mixer_applet2
4409	1	notification-ar
4411	1	clock-applet
4428	1	mapping-daemon
4431	1	gnome-screensav
4434	1	gnome-terminal
4437	4434	gnome-pty-helpe
4438	4434	bash
4716	4438	su
4719	4716	bash
4731	4719	insmod
```
</details>

I removed cache in-between commands, so that shouldn't affect my test results here.. hopefully.

Now, the next part of my issue here is - I want to make centos-2.6.18-8.1.15.el5.json.xz myself, can I just build the same kernel on the VM and run dwarf2json on the output vmlinux (with -g debug symbols) and System.map?

**Comment by ikelos** [comment]
Hiya, thanks for the detailed information!

In your first example, this seems to be the culprit:
```
Level 7  volatility3.framework.automagic.stacker: Exception during stacking: Symbol type not in LintelStacker1 SymbolTable: mnt_namespace
```
It's regrettably caused by [this line](https://github.com/volatilityfoundation/volatility3/blob/21e81c265076e1c5e21616f8db07d8c1a2699168/volatility3/framework/symbols/linux/__init__.py#L32) which attempts to change the defacto type in the symbol table, but in one this old it appears this type doesn't exist.  I'll get that fixed up.

The difficult with `-s` is somewhat covered in #618 (linxu symbols need to live under a `linux` directory within the directory that you provide to `-s`.  There's work to fix this, it's currently in #630 which no longer has that requirement (and uses a dataabase to maintain knowledge of what symbols are available and where.  The same is true trying to put the file directly into symbols, the linux.zip is essentially a zip of a `linux` subdirectory.  It must be named linux (at the moment) and any that do not live in a zip file must live under the `linux` path.  That's currently documented [here](https://volatility3.readthedocs.io/en/stable/symbol-tables.html#mac-linux-symbol-tables) in the first paragraph, bu as I say, we're looking to remove that requirement through #630, it's just waiting on review at the moment. 

We've also slightly improved [the documentation](https://volatility3.readthedocs.io/en/latest/symbol-tables.html#mac-linux-symbol-tables) in the latest development branch to give slightly more detail as to how to make your own.  Compiling your own won't result in the same banner (which includes the compile time) so this isn't the recommended way of achieving it.  Ideally you'd find the development kernel package and use that.  If not, there is some work over at the `dwarf2json` tool for creating a compilable kernel module, but it won't provide all symbols, just a selected number so it's not recommended.

Hope this helps!

**Comment by Necklaces** [comment]
@ikelos I see, thank you very much!  

I may have an issue regarding actually making the symbol table/isf file myself (instead of using the prebuilt/distributed one)  - may I create another issue regarding that?

**Comment by ikelos** [comment]
Certainly, that's fine.  5:)

**Comment by Necklaces** [comment]
Nevermind, I got it working :)

Essentially I wanted to make my own ISF file because I have a different image with a different kernel that I wanted to analyze, but I may just have been affected by this issue, which is now fixed.

Here's what I did to make my own, which might not be perfect but it works at least!

1. Get system map from https://mirror.nsc.liu.se/centos-store/5.0/os/i386/CentOS/kernel-2.6.18-8.el5.i686.rpm 

Doesn't seem like there's a debug kernel (with symbols) on that mirror, but luckily RedHat 5.0 has some for that exact kernel and compiler. 

2. Get vmlinux from https://ftp.redhat.com/pub/redhat/linux/enterprise/5Client/en/os/i686/Debuginfo/kernel-debuginfo-2.6.18-8.1.15.el5.i686.rpm

3. Run dwarf2json on those two files
4. Edit the output json to have the same banner as my memory image
5. Put output json in a linux/ folder
6. Run volatility command on image pointing to folder containing linux folder
7. ???
8. Profit!

**Comment by ikelos** [comment]
Just to note that we don't recommend people use JSON files from a non-matching kernel by manipulating the banner.  The results can't be forensically relied upon, since structure sizes can change based on the kernel configuration.

Having said all that, glad you managed to get it working!  5:)
